### DISCUSSION TITLE
Party in role vs Functional Entity

### DISCUSSION THREAD

**Discussion by NickeLind** [edmcouncil/fibo#1825] [Business Entities (BE)] [answered]
Hello, I'm trying to understand why the class [functional entity](https://spec.edmcouncil.org/fibo/ontology/BE/FunctionalEntities/FunctionalEntities/FunctionalEntity) isn't also considered a SubClass of [party in role](https://spec.edmcouncil.org/fibo/ontology/FND/Parties/Parties/PartyInRole)? I've been comparing functional entity to other SubClasses of party in role like [actor](https://spec.edmcouncil.org/fibo/ontology/FND/Parties/Parties/Actor) and I see only small differences. Is it because the property restriction of party in role [has commencement date](https://spec.edmcouncil.org/fibo/ontology/FND/Parties/Parties/hasCommencementDate) is incompatible with the top level of the SubClass hierarchy of functional entity? I know part of the functional entity SubClasses are integrated with party in role e.g [servicer](https://spec.edmcouncil.org/fibo/ontology/LOAN/LoansGeneral/Loans/Servicer). I'm particularly interested to understand if it is possible to integrate [financial service provider](https://spec.edmcouncil.org/fibo/ontology/FBC/ProductsAndServices/FinancialProductsAndServices/FinancialServiceProvider) into the party in role taxonomy.

Regards,
Niklas Lind

**Comment by NickeLind** [comment]
Hello again,
Let me rephrase the question and give your some more context to the problem I'm facing. 

I am currently working on structuring our party data aligned to FIBO (together with business owners) while collaborating with an MDM-implementation project. I've been facilitating and helping the business owners by onboarding business terms & relationships using FIBO as the primary reference model while maintaining traceability between the adjusted business terms (e.g. names) and the equivalent FIBO terms. This is done in small steps (according to a defined method and process) where the modeling efforts are driven by the current requirements in order to avoid overflowing the business owners with terms and relationships that are currently out of scope. 

We have been looking at independent parties both from the party type perspective as well as from the party in role perspective. Party in role(s), as I see it, are mostly derived from relationship between parties & agreements, parties & transactions as well as relationships between parties. When trying do describe certain entities I struggle to understand why certain classes are defined in both party in role as well as the functional entity hierarchy (example: [credit rating agency](https://spec.edmcouncil.org/fibo/ontology/FBC/DebtAndEquities/CreditRatings/CreditRatingAgency), still provisional I know).

Are party in role(s)s, due to the has commencement date property, generally supposed to be interpreted as things that are more temporary in nature than fuctional entities?

I'm right now trying to figure out if I need to introduce the whole concept of functional entities to the business in order to describe the entity [clearing house](https://spec.edmcouncil.org/fibo/ontology/FBC/FunctionalEntities/FinancialServicesEntities/ClearingHouse) or if it is fine to extend the party in role taxonomy with clearing house? Can extend the party in role taxonomy with the class-hierarchy of  [financial service provider](https://spec.edmcouncil.org/fibo/ontology/FBC/ProductsAndServices/FinancialProductsAndServices/FinancialServiceProvider) from functional entities? 

I would really love some help to understand, would it be foolish to try to integrate party in role taxonomy with functional entities (on any of the the levels proposed above) in order to keep it simple for the business owners?

Regards,
Niklas Lind

**Comment by ElisaKendall** [comment, marked as answer]
Hi Niklas,

These are great questions, and apologies for not seeing this initially.  We are using multiple classification in a number of cases, where something is both a party in role (or party role in some modeling communities) and a functional entity.  And in other cases some roles that apply to people, for example, are parties in role but not necessarily functional entities. 

There is a relatively sophisticated pattern for ownership, control, beneficial ownership, and other relationships that may not be evident at first blush - lots of property chains under the hood. The need to be able to handle the parties involved in these cases is what led to some of the modeling choices we've made. I've attached a presentation that Dean Allemang and I gave last year that talks about some of it.

[EDW2021-ModelingPatterns-v6.pdf](https://github.com/edmcouncil/fibo/files/9737532/EDW2021-ModelingPatterns-v6.pdf)

We actually have a class for clearing house - it is a subclass of financial service provider - [clearing house](https://spec.edmcouncil.org/fibo/ontology/FBC/FunctionalEntities/FinancialServicesEntities/ClearingHouse), and it has several subclasses including derivatives clearing organization, central counterparty clearing house, clearing bank, and clearing corporation. The first one is in our derivatives basics ontology and the others are defined in the same ontology as clearing house.

If it would be helpful, I'd be happy to walk through what we've done with you and answer questions on a webex or something, but my mornings are tight as you might guess. I'm based in California near San Francisco.  If you can suggest a few dates that you are available later in the evening, please send me a direct email, at ekendall@thematix.com and I'll be happy to help.

**Reply by NickeLind** [reply]
Hi Elisa,
Thanks so much for your answer! I would love to know more about FIBO, the intended use of the underlying patterns as well as the working process for contributing by proposing extensions of FIBO based on new requirements that could be relevant for the whole industry. Before I can contribute I first need to understand and learn how to correctly represent things in FIBO. Your presentation was really good and clarified some patterns regarding contextual name that are currently in scope for me. I'm right now trying to digest and understand the more complex relationships between parties, roles and situations.

Let me get back to you and mail you some dates. I would like to create some simple examples of my interpretation of functional entities as well as the modeling patterns behind the more complex relationships, using the situation class. Then maybe you could validate and correct me if I've misunderstood something? That's usually the fastest way for me to learn :)

Regards,
Niklas Lind
